### ISSUE TITLE
Wrong unit tests for aline. Aline does not function as specified.

### ISSUE THREAD

**Issue by DanielLoney** [nltk/nltk#2714]
Some of the [aline unit tests](https://github.com/nltk/nltk/blob/develop/nltk/test/unit/test_aline.py) are not correct when compared to Kondrak's original algorithms. 
For instance, in the unit test,
`aline.align('tuwθ', 'dentis') == [[('t', 'd'), ('u', 'e'), ('w', '-'), ('-', 'n'), ('-', 't'), ('-', 'i'), ('θ', 's')]]`
but the actual output according to p63 of Kondrak's Thesis is
`aline.align('tuwθ', 'dentis') == [[('t', 't'), ('uw', 'i'), '('θ', 's')]]`

I suspect the other unit tests are also written using the output of the implementation itself rather than known outputs. I am not sure what the behavior of the original python implementation is to cause this deviation, nor do I know where the source code for the original C++ implementation is to fix it. Would be nice if someone can look into this.

A list of other alignments are also given in Kondrak's Thesis which can also be used as unit tests.

(see [Kondrak's Thesis](https://webdocs.cs.ualberta.ca/~kondrak/papers/thesis.pdf), p. 63 and 151-158).

**Comment by tomaarsen** [comment]
This is a consequence of modifications that were (supposedly) made by @geoffbacon when he ported ALINE to Python. For example:
https://github.com/nltk/nltk/blob/f85fab4bb8ba59d49a03632d3f5d2535b5466913/nltk/metrics/aline.py#L183-L197

Returning to the exact functionality of Kondrak's thesis is possible - but note that it seems like a lot of small values were modified, and even some features were introduced in e.g. the `similarity_matrix` and `feature_matrix`, such as values for `labiovelar`, which I cannot find in either [Kondrak's Thesis](https://www.cs.toronto.edu/pub/gh/Kondrak-PhD-thesis-2002.pdf) or in other publications of his. (Though, perhaps I missed a more recent version with these additions)

In short, we cannot simply take the tests from Kondrak's Thesis, as our version differs slightly.

To do that, we would need to "revert" to the exact version from Kondrak's Thesis.

If we don't wish to do that, then we need to accept that we cannot make Kondrak's tests pass with our version.

**Comment by tomaarsen** [comment]
The ALINE tests were modified accordingly in cfc9461fc79943a1d6623a58b868a8f69a3c8827.
